internal/gke/cloud_patcher.go

```diff
@@ -298,6 +302,34 @@ func patchRepository(ctx context.Context, config CloudConfig, opts patchOptions,
 	}.Run(ctx)
 }
 
+// waitClusterOp waits for the given cluster operation to complete.
+func waitClusterOp(ctx context.Context, config CloudConfig, client *container.ClusterManagerClient, location string, op *containerpb.Operation) error {
+	var doneError error
+	isDone := func(op *containerpb.Operation) bool {
+		switch op.Status {
+		case containerpb.Operation_DONE:
+			return true
+		case containerpb.Operation_ABORTING:
+			doneError = fmt.Errorf("operation %s failed: %v", op.OperationType, op.Error)
+			return true
+		default:
+			return false
+		}
+	}
+	if isDone(op) {
+		return doneError
+	}
+	for r := retry.Begin(); r.Continue(ctx); {
+		if op, err := client.GetOperation(ctx, &containerpb.GetOperationRequest{
+			Name: fmt.Sprintf("projects/%s/locations/%s/operations/%s",
+				config.Project, location, op.Name),
+		}); err == nil && isDone(op) {
+			return doneError
+		}
+	}
+	return fmt.Errorf("timeout waiting for operation %s", op.OperationType)
```

Comment:
`return fmt.Errorf("timeout waiting for operation %s: %w", op.OperationType, ctx.Err())`?